File: src/test/java/com/kiwi/mobile/retrograph/RxJavaPluginsResetRule.java
package com.kiwi.mobile.retrograph;

import org.junit.rules.*;
import org.junit.runner.*;
import org.junit.runners.model.*;

import io.reactivex.plugins.*;

final class RxJavaPluginsResetRule implements TestRule {
  @Override
  public Statement apply(final Statement base, Description description) {
    return new Statement() {
      @Override
      public void evaluate() throws Throwable {
        RxJavaPlugins.reset();
        try {
          base.evaluate();
        } finally {
          RxJavaPlugins.reset();
        }
      }
    };
  }
}

File: src/test/java/com/kiwi/mobile/retrograph/rule/RxJavaPluginsResetRule.kt
package com.kiwi.mobile.retrograph.rule

import io.reactivex.plugins.*

import org.junit.rules.*
import org.junit.runner.*
import org.junit.runners.model.*

class RxJavaPluginsResetRule:
  TestRule {

  override fun apply(base: Statement, description: Description) =
    object: Statement() {

      @Throws(Throwable::class)
      override fun evaluate() {
        RxJavaPlugins.reset()
        try {
          base.evaluate()
        } finally {
          RxJavaPlugins.reset()
        }
      }
    }
}
